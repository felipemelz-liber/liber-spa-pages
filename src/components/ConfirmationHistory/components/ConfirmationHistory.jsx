import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import ToastContainer from 'liber-components/components/Toast';
import { fetchConfirmations, setCredentials, updateFilters } from '../actions/index.js';
import ConfirmationHistoryTable from './Table.jsx';
import {
  ConfirmationHistoryContainer,
  LoadingBlock,
  Loading,
  Global,
  ScreenTitle,
  TitleContainer,
} from './ConfirmationHistory.styles.jsx';

const DATE_LOWER = 'q[created_at_date_lteq]';
const DATE_GREATER = 'q[created_at_date_gteq]';
const DATE_FORMAT = 'YYYY-MM-DD';

export const ConfirmationHistory = ({
  token,
  sacadoId,
  onSetCredentials,
  onFetchConfirmations,
  loadingAllPage,
  per,
  onUpdateFilters,
}) => {
  React.useEffect(() => {
    onSetCredentials(token, sacadoId);
    onFetchConfirmations({ token, params: { page: '1', per: '10' } });
  }, []);

  const handleFilter = ({ start, end }) => {
    let newFilters = {};
    if (start && end) {
      newFilters = {
        [DATE_GREATER]: start.format(DATE_FORMAT),
        [DATE_LOWER]: end.format(DATE_FORMAT),
      };
    } else if (start || end) {
      newFilters = {
        [DATE_GREATER]: start ? start.format(DATE_FORMAT) : end.format(DATE_FORMAT),
        [DATE_LOWER]: start ? start.format(DATE_FORMAT) : end.format(DATE_FORMAT),
      };
    }
    onUpdateFilters(newFilters);
    onFetchConfirmations({ token, params: { page: '1', per, ...newFilters } });
  };

  return (
    <>
      <Global />
      {loadingAllPage ? <Loading loop /> : null}
      <LoadingBlock loading={loadingAllPage ? 1 : 0}>
        <ConfirmationHistoryContainer>
          <TitleContainer>
            <ScreenTitle>Histórico de Confirmações</ScreenTitle>
          </TitleContainer>
          <ConfirmationHistoryTable handleFilter={handleFilter} />
        </ConfirmationHistoryContainer>
      </LoadingBlock>
      <ToastContainer />
    </>
  );
};

ConfirmationHistory.propTypes = {
  onSetCredentials: PropTypes.func.isRequired,
  token: PropTypes.string.isRequired,
  sacadoId: PropTypes.number.isRequired,
  onFetchConfirmations: PropTypes.func.isRequired,
  loadingAllPage: PropTypes.bool.isRequired,
  per: PropTypes.number.isRequired,
  onUpdateFilters: PropTypes.func.isRequired,
};

const mapStateToProps = ({ confirmationHistory }) => ({
  loadingAllPage: confirmationHistory.loadingAllPage,
  per: confirmationHistory.pagination.per,
});

const mapDispatchToProps = {
  onFetchConfirmations: fetchConfirmations,
  onSetCredentials: setCredentials,
  onUpdateFilters: updateFilters,
};

export default connect(mapStateToProps, mapDispatchToProps)(ConfirmationHistory);
