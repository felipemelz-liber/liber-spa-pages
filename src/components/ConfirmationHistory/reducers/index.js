import confirmationHistory from './ConfirmationHistoryReducer';

export default {
  confirmationHistory,
};
