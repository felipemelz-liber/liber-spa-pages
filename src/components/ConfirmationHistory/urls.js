const domain = INVESTOR_DASHBORD_API; // eslint-disable-line

export const CONFIRMATIONS_HISTORY_URL = `${domain}/api/v1/sacado/direito_creditorios`;
