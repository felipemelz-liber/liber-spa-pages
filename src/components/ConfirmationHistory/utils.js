import React from 'react';
import PropTypes from 'prop-types';
import { StyledButton, DownloadIcon } from './components/Table.styles.jsx';

export const filtersPropTypes = PropTypes.shape({
  'q[created_at_eq]': PropTypes.string,
  'q[created_at_gteq]': PropTypes.string,
  'q[created_at_lteq]': PropTypes.string,
});

export const CONFIRMATION_HIST_COLUMNS = [
  {
    name: 'slug',
    label: 'Código',
    width: 12,
    align: 'center',
  },
  {
    name: 'value',
    label: 'Valor Total',
    width: 18,
    align: 'right',
    getCellValue: ({ value }) => `R$ ${value}`,
  },
  {
    name: 'created_at',
    label: 'Data de Criação',
    width: 18,
    align: 'right',
  },
  {
    name: 'contract',
    label: 'Contrato',
    width: 25,
    align: 'left',
    // eslint-disable-next-line react/prop-types
    getCellValue: ({ contract = '#' }) => (
      <StyledButton version={2} size="small" href={contract} download>
        <DownloadIcon />
        Download
      </StyledButton>
    ),
  },
];
