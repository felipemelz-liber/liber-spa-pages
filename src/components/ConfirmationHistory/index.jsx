import React from "react";
import PropTypes from "prop-types";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import { LiberV4 } from "liber-components";
import hydrateStore from "../store";
import ConfirmationHistory from "./components/ConfirmationHistory.jsx";

const View = ({ token, sacadoId }) => (
  <>
    <Provider store={hydrateStore({})}>
      <ConfirmationHistory token={token} sacadoId={sacadoId} />
    </Provider>
  </>
);

export default View;
