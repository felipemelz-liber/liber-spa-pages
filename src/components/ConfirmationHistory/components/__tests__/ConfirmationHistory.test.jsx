import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import { ConfirmationHistory } from '../ConfirmationHistory';

describe('Testing ConfirmationHistory component', () => {
  configure({ adapter: new Adapter() });
  const props = {
    token: 'token',
    sacadoId: 1,
    onSetCredentials: sinon.spy(),
    onFetchConfirmations: sinon.spy(),
    loadingAllPage: false,
    per: 10,
    onUpdateFilters: () => null,
  };
  it('ConfirmationHistory should render correctly', () => {
    const useEffectStub = sinon.stub(React, 'useEffect');
    useEffectStub.callsArg(0);

    const wrapper = shallow(<ConfirmationHistory {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();

    const { onSetCredentials, onFetchConfirmations } = props;
    expect(onSetCredentials.called).toBeTruthy();
    expect(onFetchConfirmations.called).toBeTruthy();
  });
});
