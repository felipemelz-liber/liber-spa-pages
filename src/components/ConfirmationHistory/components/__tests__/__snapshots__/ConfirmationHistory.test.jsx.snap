// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ConfirmationHistory component ConfirmationHistory should render correctly 1`] = `
<Fragment>
  <GlobalStyleComponent />
  <styled.div
    loading={0}
  >
    <styled.div>
      <styled.div>
        <styled.span>
          Histórico de Confirmações
        </styled.span>
      </styled.div>
      <Connect(ConfirmationHistoryTable)
        handleFilter={[Function]}
      />
    </styled.div>
  </styled.div>
  <_default />
</Fragment>
`;
